### PR TITLE
Add "auth|security:checkRights" API actions

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Adds `auth:checkRights` and `security:checkRights` API actions to check if an API request can be executed by the current logged user or by any user.

### CODE REVIEW

File: doc/2/api/controllers/auth/check-rights/index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+code: true
+type: page
+title: checkRights
+---
+
+# checkRights
+
+<SinceBadge version="change-me"/>
```
Comment (jenow) [inline doc/2/api/controllers/auth/check-rights/index.md:9]: Is this normal?

File: doc/2/api/controllers/security/check-rights/index.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+code: true
+type: page
+title: checkRights
+---
+
+# checkRights
+
+<SinceBadge version="change-me"/>
```
Comment (jenow) [inline doc/2/api/controllers/security/check-rights/index.md:9]: Is this normal?

**Review by jenow — APPROVED** [review]

File: doc/2/api/controllers/auth/check-rights/index.md (hunk 1 above)
Comment (Aschen) [reply doc/2/api/controllers/auth/check-rights/index.md:9]: Yes, the release tool will replace this by the correct version number when creating the release

File: doc/2/api/controllers/security/check-rights/index.md (hunk 2 above)
Comment (Aschen) [reply doc/2/api/controllers/security/check-rights/index.md:9]: Yes, the release tool will replace this by the correct version number when creating the release

File: doc/2/api/controllers/auth/check-rights/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+```http
+URL: http://kuzzle:7512/_checkRights
+Method: POST
+Body:
+```
+
+```js
+{
+  "request": {
```
Comment (scottinet) [inline doc/2/api/controllers/auth/check-rights/index.md:27]: Is this sublevel necessary?

File: doc/2/api/controllers/auth/check-rights/index.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+```js
+{
+  "status": 200,
+  "error": null,
+  "controller": "auth",
+  "action": "checkRights",
+  "requestId": "<unique request identifier>",
+  "result": {
+    "allowed": true
+  }
```
Comment (scottinet) [inline doc/2/api/controllers/auth/check-rights/index.md:84]: same here, can't we have only a boolean result?

File: doc/2/api/controllers/security/check-rights/index.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+Checks if the provided API request can be executed by an user.
```
Comment (scottinet) [inline doc/2/api/controllers/security/check-rights/index.md:11]:
```suggestion
Checks if the provided API request can be executed by a user.
```

File: doc/2/api/controllers/auth/check-rights/index.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+code: true
+type: page
+title: checkRights
+---
+
+# checkRights
+
+<SinceBadge version="change-me"/>
+
+Checks if the provided API request can be executed.
```
Comment (scottinet) [inline doc/2/api/controllers/auth/check-rights/index.md:11]: (open to debate) I think a bit more context would help.

```suggestion
Checks if the provided API request can be executed by this network connection, using the current authentication information.
```

File: doc/2/api/controllers/security/check-rights/index.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+## Arguments
+
+- `_id`: user unique [kuid](/core/2/guides/essentials/user-authentication#kuzzle-user-identifier-kuid)
+
+---
+
+## Body properties
+
+- `request`: API request to check the rights with
```
Comment (scottinet) [inline doc/2/api/controllers/security/check-rights/index.md:70]:
```suggestion
- `request`: API request to check
```

File: doc/2/api/controllers/auth/check-rights/index.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  }
+}
+```
+
+---
+
+## Body properties
+
+- `request`: API request to check the rights with
```
Comment (scottinet) [inline doc/2/api/controllers/auth/check-rights/index.md:63]:
```suggestion
- `request`: API request to check
```

**Review by scottinet — CHANGES_REQUESTED** [review]

File: doc/2/api/controllers/auth/check-rights/index.md (hunk 3 above)
Comment (Aschen) [reply doc/2/api/controllers/auth/check-rights/index.md:27]: Not necessarily, I wanted to be clear that users should pass a request but I don't have string feeling about having the sublevel or not

File: doc/2/api/controllers/auth/check-rights/index.md (hunk 4 above)
Comment (Aschen) [reply doc/2/api/controllers/auth/check-rights/index.md:84]: If we simply return a boolean result then we will never be able to change this action return without a breaking change.

**Review by scottinet — APPROVED** [review]

**Pull request merged**